cloudflare/resource_cloudflare_logpush_job.go

```diff
@@ -34,6 +35,11 @@ func resourceCloudflareLogpushJob() *schema.Resource {
 				Type:     schema.TypeString,
 				Optional: true,
 			},
+			"dataset": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringInSlice([]string{"http_requests", "spectrum_events"}, false),
```

Comment:
The [API documentation](https://api.cloudflare.com/#logpush-jobs-properties) has the validation has alphanumeric characters only however only "http_requests" and "spectrum_events" are listed in the developer guide. This leads me to believe that there are more data sets coming but for now we can be specific here and catch some user error before it gets too far into the processing to identify the issue.